File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/repository/CollectTagRepository.java
package com.windea.demo.cloudcollect.core.repository;

import com.windea.demo.cloudcollect.core.domain.entity.CollectTag;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface CollectTagRepository extends JpaRepository<CollectTag, Long> {
	Optional<CollectTag> findByName(String name);

	Page<CollectTag> findByUser_Id(Long userId, Pageable pageable);

	long countByUser_Id(Long userId);

	Page<CollectTag> findByUser_IdAndNameContains(Long userId, String name, Pageable pageable);

	boolean existsByUser_IdAndName(Long userId, String name);
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/repository/CollectTagRepository.kt
package com.windea.demo.cloudcollect.core.repository

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*
import org.springframework.data.jpa.repository.*

interface CollectTagRepository : JpaRepository<CollectTag, Long> {
	fun findByNameAndUserId(name: String, userId: Long): CollectTag?
	
	fun findAllByNameContainsAndUserId(name: String, userId: Long, pageable: Pageable): Page<CollectTag>
	
	fun findAllByUserId(userId: Long, pageable: Pageable): Page<CollectTag>
	
	fun countByUserId(userId: Long): Long
	
	fun existsByNameAndUser(name: String, user: User): Boolean
}
